functorch/csrc/DynamicLayer.cpp

```diff
@@ -551,21 +548,20 @@ void dynamicLayerFrontFallback(const c10::OperatorHandle& op, torch::jit::Stack*
 
   // Re-dispatch
   op.callBoxed(stack);
-  if (called_functionalize_kernel) {
-    auto ret_size = op.schema().returns().size();
-    foreachTensorInplace(*stack, stack->size() - ret_size, stack->size(),
-      [&](const Tensor& tensor) {
-        TORCH_INTERNAL_ASSERT(at::functionalization::impl::isFunctionalTensor(tensor));
+  auto ret_size = op.schema().returns().size();
+  foreachTensorInplace(*stack, stack->size() - ret_size, stack->size(),
+    [&](const Tensor& tensor) {
+      if (at::functionalization::impl::isFunctionalTensor(tensor)) {
         auto wrapper = at::functionalization::impl::unsafeGetFunctionalWrapper(tensor);
         // Functorch is responsible for setting the level on the wrapper, since we don't
         // have that info available in core (for now).
         // We could just "propagate" the level from the input tensors inside of the functionalize kernels,
         // but unfortunately we can't do that for factory operators.
         wrapper->set_level(layer.layerId());
-        return tensor;
       }
-    );
-  }
+      return tensor;
```

Comment:
Apparently you need to compile with the `-Werror=return-type` flag, or else you get silent UB if you forget to return from a non-void function (aka this bug) :(

Reply:
:(, we used to have -Werror turned on for all errors but it turns out PyTorch doesn't and PyTorch would introduce warnings